eng/pipelines/templates/steps/template-test-run-job.yml

```diff
@@ -43,27 +48,135 @@ steps:
     Write-Host "##vso[task.setvariable variable=CleanupImmediate]$CleanupImmediate"
   displayName: Set test run parameters
 
-- task: Bash@3
-  displayName: Test templates
-  env:             
-    # Required secrets for Terraform service principal authentication
-    ARM_CLIENT_ID: $(arm-client-id)
-    ARM_CLIENT_SECRET: $(arm-client-secret)
-    ARM_TENANT_ID: $(arm-tenant-id)
+- pwsh: |
+    git config --global user.name "azure-sdk"
+    git config --global user.email "azuresdk@microsoft.com"
+    echo "https://azure-sdk:$(azuresdk-github-pat)@github.com" > ~/.git-credentials
```

Comment:
do we need to use `env:` to map the `azuresdk-github-pat` ?  @danieljurek

Reply:
We don't need `env` in this case. If the secret is associated in the pipeline the script will be generated with the secret written in... 

@hemarina -- Use this template instead of duplicating the git credential setting code: https://github.com/Azure/azure-dev/blob/main/eng/pipelines/templates/steps/set-git-credentials.yml

You can include it with: 

```yaml
- template: /eng/pipelines/templates/steps/set-git-credentials.yml
```

You'll need to break the `git checkout` into a separate task.